CIS crashes when a wrong ConfigMap is created and then deleted

#### Setup Details
 CIS Version : 2.7.1
 Build: f5networks/k8s-bigip-ctlr:latest       
 BIGIP Version: Big IP x.x.x      
 AS3 Version: 3.x      
 Agent Mode: AS3/CCCL       
 Orchestration: K8S/OSCP      
 Orchestration Version:       
 Pool Mode: Cluster/Nodeport      
 Additional Setup details: <Platform/CNI Plugins/ cluster nodes/ etc>

#### Description  
CIS crashes when a "wrong" ConfigMap is added and then delete

#### Steps To Reproduce
1) Create namespace monitoring
2) Apply the following ConfigMap.
3) Delete the configmap

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: f5-as3-declaration
  namespace: monitoring
  labels:
    f5type: virtual-server
    as3: "true"
data:
  template: |
    {
        "class": "AS3",
        "action": "deploy",
        "persist": false,        
        "declaration": {
            "class": "ADC",
            "schemaVersion": "3.28.0",
            "id": "f5-as3-sensu-system",
            "label": "f5-as3-sensu-system",
            "remark": "AS3 Declaration - NS sensu-system",
            "sensu-system_infs_tenant": {
                "class": "Tenant",
                "Shared": {
                    "class": "Application",
                    "template": "shared",
                    "alertmanager-cluster_pool": {
                        "class": "Pool",
                        "monitors": [
                            "tcp"
                        ],
                        "members": [
                            {
                                "servicePort": 9094,
                                "serverAddresses": [],
                                "shareNodes": true
                            }
                        ]
                    }
                },                
                "alertmanager-cluster_tcp": {
                    "class": "Application",
                    "template": "generic",
                    "alertmanager-cluster_tcp": {
                        "class": "Service_TCP",
                        "remark": "alertmanager.x.x.x",
                        "virtualAddresses": [
                            {
                                "10.10.10.10"
                            }
                        ],
                        "virtualPort": 9094,
                        "allowVlans": [
                            "/Common/VL-lb-int"
                        ],         
                        "snat": "auto",                        
                        "pool": "/sensu-system_infs_tenant/Shared/alertmanager-cluster_pool"
                    }
                }
            }
        }
    }
```

#### Actual Result 
The logs from CIS were the following

022/02/11 05:33:32 [DEBUG] [CORE] Finished syncing virtual servers  in namespace monitoring (129.525µs), 21/21
E0211 05:33:32.750478       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 179 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x195ad40, 0x28dcc30)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:74 +0x95
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:48 +0x89
panic(0x195ad40, 0x28dcc30)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncConfigMaps(0xc0002ae000, 0xc0006e5908, 0xc0000d2520, 0xa, 0x0, 0x0, 0x1b84bc7, 0xa, 0xc0007c4240, 0x12, ...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1355 +0x2318
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncVirtualServer(0xc0002ae000, 0xc0000d2520, 0xa, 0x0, 0x0, 0x1b84bc7, 0xa, 0xc0007c4240, 0x12, 0x1b81442, ...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1302 +0xac5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).processNextVirtualServer(0xc0002ae000, 0x203000)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1149 +0x1b8
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).virtualServerWorker(0xc0002ae000)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1025 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00068ac50)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00068ac50, 0x1d74480, 0xc000758240, 0x1, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00068ac50, 0x3b9aca00, 0x0, 0x100000000000001, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc00068ac50, 0x3b9aca00, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).runImpl
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:969 +0x130
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x16ade78]

goroutine 179 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:55 +0x10c
panic(0x195ad40, 0x28dcc30)
        /usr/local/go/src/runtime/panic.go:969 +0x1b9
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncConfigMaps(0xc0002ae000, 0xc0006e5908, 0xc0000d2520, 0xa, 0x0, 0x0, 0x1b84bc7, 0xa, 0xc0007c4240, 0x12, ...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1355 +0x2318
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).syncVirtualServer(0xc0002ae000, 0xc0000d2520, 0xa, 0x0, 0x0, 0x1b84bc7, 0xa, 0xc0007c4240, 0x12, 0x1b81442, ...)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1302 +0xac5
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).processNextVirtualServer(0xc0002ae000, 0x203000)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1149 +0x1b8
github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).virtualServerWorker(0xc0002ae000)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:1025 +0x2b
k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc00068ac50)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:155 +0x5f
k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc00068ac50, 0x1d74480, 0xc000758240, 0x1, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:156 +0xad
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00068ac50, 0x3b9aca00, 0x0, 0x100000000000001, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:133 +0x98
k8s.io/apimachinery/pkg/util/wait.Until(0xc00068ac50, 0x3b9aca00, 0xc0000921e0)
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:90 +0x4d
created by github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager.(*Manager).runImpl
        /go/src/github.com/F5Networks/k8s-bigip-ctlr/pkg/appmanager/appManager.go:969 +0x130

--- a/docs/RELEASE-NOTES.rst
+++ b/docs/RELEASE-NOTES.rst
@@ -21,7 +21,9 @@ Enhancements
 * Added fix for cis crash with routes
 * :issues:`2212` Fix ExternalDNS adds both VSs to a Wide IP pool with using "httpTraffic: allow" with VS CR
 * :issues:`2222` Fix deleting VirtualServer using hostGroup
-* :issues:`2233` TS and VS CRD don't detect the pool members
+* :issues:`2233` TS and VS CRD don't detect the pool members for grafana service
+* :issues:`2234` Fix for CIS crash with subsequent creation and deletion of wrong ConfigMap
+
 
 2.7.1
 -------------

--- a/pkg/appmanager/appManager.go
+++ b/pkg/appmanager/appManager.go
@@ -1415,9 +1415,12 @@ func (appMgr *Manager) syncConfigMaps(
 	if appMgr.AgentCIS.IsImplInAgent(ResourceTypeCfgMap) {
 		key := sKey.Namespace + "/" + sKey.ResourceName
 		if sKey.Operation == OprTypeDelete && sKey.ResourceKind == Configmaps {
-			appMgr.agentCfgMap[key].Operation = OprTypeDelete
-			stats.vsDeleted += 1
+			if _, ok := appMgr.agentCfgMap[key]; ok {
+				appMgr.agentCfgMap[key].Operation = OprTypeDelete
+				stats.vsDeleted += 1
+			}
 			return nil
+
 		}
 		if nil != svc {
 			tntLabel, tntOk := svc.ObjectMeta.Labels["cis.f5.com/as3-tenant"]

